taskipy/task_runner.py

```diff
@@ -93,14 +97,36 @@ def send_signal_to_task_process(signum: int, _frame) -> None:
 
         return process.returncode
 
-    def __pre_task(self, task_name: str):
-        return self.__find_hooks('pre', task_name)
+    def __resolve_variables(self, command: str) -> str:
+        types = self.__get_variable_types(self.__project.variables)
+        resolved_vars = types["nonrecursive"]
+        recursive_vars = types["recursive"]
 
-    def __post_task(self, task_name: str):
-        return self.__find_hooks('post', task_name)
+        while len(recursive_vars) > 0:
+            resolved_count = len(resolved_vars)
 
-    def __find_hooks(self, hook_type: str, task_name: str) -> Optional[Task]:
-        try:
-            return self.__project.tasks[f'{hook_type}_{task_name}']
-        except KeyError:
-            return None
+            for name, value in recursive_vars.copy().items():
+                try:
+                    resolved_vars[name] = value.format(**resolved_vars)
+                    recursive_vars.pop(name)
+                except KeyError:
+                    pass
+
+            if resolved_count == len(resolved_vars):
+                raise CircularVariableError()
+
+        return command.format(**resolved_vars)
+
+    def __get_variable_types(self, variables: Dict[str, Variable]) -> Dict[str, Dict[str, str]]:
```

Comment:
suggestion: change this method to return a tuple: (nonrecursive, recursive).

reason: we always return nonrecursive and recursive. If we use a dict, it means that we need to trust the developer to remember that these are the only two keys in the dict (no way to type that). If we use a tuple, we can fully type the return value and so ensure that everyone will always use the following:

```python
nonrecursive_vars, recursive_vars = self.__get_variable_types(vars)
```

wdyt?

Reply:
I agree. This is better